scripts/mysql_system_tables_fix.sql

```diff
@@ -1002,62 +1002,213 @@ INSERT IGNORE INTO mysql.global_grants VALUES ('mysql.session', 'localhost', 'SE
 # upgrade.
 INSERT IGNORE INTO mysql.global_grants VALUES ('mysql.session', 'localhost', 'SYSTEM_USER', 'N');
 
+set @is_mysql_encrypted = (select ENCRYPTION from information_schema.INNODB_TABLESPACES where NAME='mysql');
+
+SET @str="ALTER TABLE mysql.db ENCRYPTION='Y'";
+SET @cmd = IF(STRCMP(@is_mysql_encrypted,'Y'), 'SET @dummy = 0', @str);
+PREPARE stmt FROM @cmd;
+EXECUTE stmt;
+DROP PREPARE stmt;
+
 # Move all system tables with InnoDB storage engine to mysql tablespace.
 SET @cmd="ALTER TABLE mysql.db TABLESPACE = mysql";
 PREPARE stmt FROM @cmd;
 EXECUTE stmt;
 DROP PREPARE stmt;
 
+SET @str="ALTER TABLE mysql.user ENCRYPTION='Y'";
```

Comment:
We should avoid modifying all create table statements for mysql database. Instead do ALTER DATABASE mysql default encryption='y' if mysql tablespace is encrypted.

Reply:
From documentation:
Changing the default encryption of the database does not alter the encryption of existing tables associated with the schema. Only newly-created tables inherit the default database encryption.

Thus I need to set encryption explicitly for all tables inside mysql schema.

Reply:
Yes, if this can be set before tables are created, we can avoid the explicit change for all create tables.

Reply:
but this is script for upgrade - so it works on already existing tables.